controllers/grid/users/chapter/form/ChapterForm.inc.php

```diff
@@ -162,12 +163,24 @@ public function fetch($request, $template = null, $display = false)
         $selectedChapterFiles = [];
 
         if ($this->getChapter()) {
-            $selectedChapterAuthors = DAORegistry::getDAO('ChapterAuthorDAO')->getAuthors($this->getPublication()->getId(), $this->getChapter()->getId())->toArray();
+            $selectedChapterAuthors = Repo::author()->getMany(
+                Repo::author()
+                    ->getCollector()
+                    ->filterByChapter($this->getChapter()->getId())
+                    ->filterByPublicationIds([$this->getPublication()->getId()])
+            );
+
             foreach ($selectedChapterAuthors as $selectedChapterAuthor) {
                 $chapterAuthorOptions[$selectedChapterAuthor->getId()] = $selectedChapterAuthor->getFullName();
             }
         }
-        $authorsIterator = Services::get('author')->getMany(['publicationIds' => $this->getPublication()->getId(), 'count' => 1000]);
+        $authorsIterator = Repo::author()->getMany(
+            Repo::author()
+                ->getCollector()
+                ->filterByPublicationIds([$this->getPublication()->getId()])
+                ->limit(1000)
```

Comment:
You can remove the `->limit(1000)` here. That's an old convention I put in place but we decided to abandon it.

Reply:
Fixed